src/test/java/com/dakar/dakar/integration/AnnotationControllerTest.java

```diff
@@ -0,0 +1,76 @@
+package com.dakar.dakar.integration;
+
+import com.dakar.dakar.controller.AnnotationController;
+import com.dakar.dakar.models.Journey;
+import com.dakar.dakar.services.implementation.JourneyServiceImpl;
+import com.dakar.dakar.services.interfaces.IJourneyService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.web.reactive.function.BodyInserters;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class AnnotationControllerTest extends AbstractIntegrationTest{
+
+    @Autowired
+    private WebTestClient webClient;
+
+    Flux<Journey> flux = Flux.just(
+            new Journey("Jack", "Bauer", "pompei"),
+            new Journey("Chloe", "O'Brian", "afghanistan"),
+            new Journey("afghanistan", "Bauer", "chicago"),
+            new Journey("David", "Palmer", "dubai"),
+            new Journey("Michelle", "Dessler", "portugal"));
+
+    @Test
+    public void testRouteWithAnnotationHateoasWithAssembler() {
+        this.webClient.get().uri("/test2/afghanistan")
+                .exchange()
```

Comment:
I don't get what does the exchange method, could you explain me during the next call ?

Reply:
It allows to get the response